addons/mail/models/mail_channel.py

```diff
@@ -746,6 +747,36 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def toggle_message_pin(self, message_id, pinned):
+        """ (Un)pin a message on the channel and send a notification to the
+        members.
+        :param message_id: id of the message to be pinned.
+        :param pinned: whether the message should be pinned or unpinned.
+        """
+        self.ensure_one()
+        if not self.env["mail.message"].search([
+            ['id', '=', message_id],
+            ['pinned_on_channel_id', '=', None if pinned else self.id]
+        ]):
+            return
+        self.write({
+            'pinned_message_ids': [Command.link(message_id) if pinned else Command.unlink(message_id)]
+        })
+        self.env['bus.bus']._sendone(self, 'mail.record/insert', {
+            'Message': {
+                'id': message_id,
+                'pinned': pinned,
+            }
+        })
+        if pinned:
+            notification = _(f'''
+                <div data-oe-type="pin" class="o_mail_notification">
+                    %s pinned a <a href="#" data-oe-model="highlight" data-oe-id="{message_id}">message</a> to this channel.
+                    <a href="#" data-oe-model="pin-menu">See all pinned messages.</a>
+                </div>
```

Comment:
I would just put `oe-type` on all of them, for named actions, whether `pin`, `highlight` or `pin-menu`